### PR TITLE
add transformer

### PR DESCRIPTION
**Pull request by bradyhhhuang**
I tried to build a decoder only transformer model with masked attention and positional encoding.
However, the training loss stucks at around 0.05 which is not low enough(compared to 0.005 in lstm )
Also the validation loss is weirdly low, and the comparison plot is showing that my model predicts nothing but 0.04...
So I guess there should be some problem with my model structure.

One thing needs to be improved is that my model structure is not able to run in GPU, need some modifications.

### CODE REVIEW

**Review by bytehut — APPROVED** [review]
- whats the feature dimension you are passing in? maybe frame_len should be larger than 1 since I believe transformers perform better with larger dimension. dim_model could be 128, and frame_len could be dim_model/2, or we can add a linear layer to project our frame_len into dim_model for more flexibility
- should we be passing in the entire sequence at once? currently we are passing in the data the same way as LSTM. no need for truncated backprop, initializing hidden state, etc
